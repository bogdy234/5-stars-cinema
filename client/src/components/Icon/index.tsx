import { FC, ReactElement } from "react";

interface IconProps {
    src: string;
    alt: string;
    className?: string;
    onClick?: () => void;
}

const Icon: FC<IconProps> = ({
    src,
    alt,
    className,
    onClick,
}: IconProps): ReactElement => {
    return (
        <div className={`${className}`}>
            <img
                src={src}
                alt={alt}
                className={`w-full h-full ${onClick && "cursor-pointer"}`}
                onClick={onClick}
            />
        </div>
    );
};

export default Icon;
