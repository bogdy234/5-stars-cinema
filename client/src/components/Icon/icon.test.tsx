import renderer from "react-test-renderer";
import Icon from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<Icon src={"/logout.svg"} alt={"some-alt"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
