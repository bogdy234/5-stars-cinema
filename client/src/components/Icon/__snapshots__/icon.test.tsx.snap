// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="undefined"
>
  <img
    alt="some-alt"
    className="w-full h-full undefined "
    src="/logout.svg"
  />
</div>
`;
